### PR TITLE
Cup

### PR DESCRIPTION
**Pull request by teelu**
basic cup constructors and functions and created a for loop to create cups

### CODE REVIEW

**Comment by teelu** [comment]
fixing this PR, do not review till messaged

**Comment by teelu** [comment]
Code is fine to review but not linted.

File: js/application.js [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+  for (var i = 0; i < 3; i++) {
+    cupArray.push(new cup("cup" + eval('i+1'),i,false));
+  }
+  console.log(cupArray);
+  //Pick a cup
+  console.log("Where to insert ball");
+  var cupSelected = prompt();
+  //set cup selected
+  cupArray[cupSelected].hasBall = true;
 });
```
Comment (h8rry) [inline js/application.js]: Always include a line break at the end of each file

File: js/application.js [hunk 2]
```diff
@@ -1,3 +1,21 @@
+'use strict';
 $(document).ready(function() {
-	console.log("READY");
+  //cup holder(har har har)
+  var cupArray = [];
+  //cup functions
+  function cup(name,position,hasBall) {
```
Comment (h8rry) [inline js/application.js]: Need one space after `,`

**Comment by h8rry** [comment]
Can merge when fixes are done

**Comment by teelu** [comment]
Added line breaks and changes  (forgot about rebase so 2 commits sorry X_X)

File: js/application.js [hunk 3]
```diff
@@ -1,3 +1,22 @@
+'use strict';
 $(document).ready(function() {
-	console.log("READY");
-});
\ No newline at end of file
+  //cup holder(har har har)
+  var cupArray = [];
+  //cup functions
+
+  function cup(name, position, hasBall) {
```
Comment (f3r) [inline js/application.js]: Convention is to use `Cup` instead of `cup`

**Comment by f3r** [comment]
Food for thought...

``` js
function Cup(name, position, hasBall) {
  this.name = name;
  this.position = position;
  this.hasBall = hasBall || false;
}

//generate cups
var cups = [1,1,1].map(function(val, index){ 
  return new Cup('cup'+(index+1), index)
})
```

**Comment by teelu** [comment]
changed cup to Cup.  
Thanks for the other way of declaring the array.  It definitely looks cooler than the way I do it!

**Pull request merged**